pulp_rpm/app/models.py

```diff
@@ -678,3 +681,43 @@ class RpmDistribution(PublicationDistribution):
     """
 
     TYPE = 'rpm'
+
+
+class Modulemd(Content):
+    """
+    The "Modulemd" content type. Modularity support.
+
+    Fields:
+        name (Text):
+            Name of the modulemd
+        stream (Text):
+            The modulemd's stream
+        version (Text):
+            The version of the modulemd.
+        context (Text):
+            The context flag serves to distinguish module builds with the
+            same name, stream and version and plays an important role in
+            future automatic module stream name expansion.
+        arch (Text):
+            Contains a string describing the module's artifacts' main hardware
+            architecture compatibility.
+        dependencies (Text):
+            Module dependencies, if any.
+        artifacts (List):
+            Artifacts shipped with this module.
+        _packages (List):
+            List of RPM Packages connected to this modulemd.
+    """
+
+    TYPE = "modulemd"
+
+    # required metadata
+    name = models.CharField(max_length=255)
+    stream = models.CharField(max_length=255)
+    version = models.CharField(max_length=255)
+    context = models.CharField(max_length=255)
+    arch = models.CharField(max_length=255)
+
+    dependencies = models.TextField(default='[]')
+    artifacts = models.TextField(default='[]')
+    _packages = models.ManyToManyField(Package)
```

Comment:
@dralley, I understand that you want to make clear that `_packages` are not part of modulemd metadata. However, I think we had a convention that fields which start with underscore come from the pulpcore. Do I remember it correctly?
What about other models? E.g. this PR add a modular flag to the Package model, should it start also with underscore?

Maybe we can just define fields which don't come from metadata separately ( new line + comment that those fields are not part of metadata)? Is it important for the field/attribute name to be clear that it's a part of metadata or not?

Reply:
> I understand that you want to make clear that _packages are not part of modulemd metadata. However, I think we had a convention that fields which start with underscore come from the pulpcore. Do I remember it correctly?

You remember correctly, however it doesn't necessarily have to be that specific -- maybe we want it to mean "not part of the metadata of the type" which would cover both cases.

I could go either way on this to be honest, I don't have a strong feeling about it.